### PR TITLE
Android: Remove Migration Code

### LINKED ISSUE #13540: Android: Remove migration code and external storage permission
##### Minetest version

```
since 5.5.0 to latest workflow build (master)
```

##### OS / Hardware

Operating system: Android 6.0.1 (API lv 23)
CPU: MediaTek

GPU model: Mali
OpenGL ES version: 3.1

##### Summary

since 5.5.0 Minetest Android use data folder for files instead of Minetest directory but Minetest still request storage permission
##### Steps to reproduce

simply run Minetest Android

### PR DESCRIPTION
**Pull request by oong819**
Add compact, short information about your PR for easier understanding:

- Goal of the PR
Remove Migration Code
- How does the PR work?
Remove Migration Code
- Does it resolve any reported issue?
This PR fixes #13540.
- Does this relate to a goal in the roadmap?
not yet checked
- If not a bug fix, why is this PR needed? What usecases does it solve?

less permission, less less code
make Minetest safer to run and free up some memory used for storing permissions stuff

## To do

This PR is Ready for Review.

## How to test

1. Open Minetest.
2. Things should work fine.

-------

Edit by srifqi: Add the reported issue

### CODE REVIEW

**Comment by oong819** [comment]
screwed up the code

**Comment by oong819** [comment]
un screwed up the code

**Comment by oong819** [comment]
i found  why its  failed, fixing

**Comment by oong819** [comment]
forgot a semicolon haha

**Comment by oong819** [comment]
ITS DONE REVIEW PLS?

File: android/app/src/main/java/net/minetest/minetest/MainActivity.java [hunk 1]
```diff
@@ -100,57 +97,10 @@ public void onCreate(Bundle savedInstanceState) {
 		mProgressBar = findViewById(R.id.progressBar);
 		mTextView = findViewById(R.id.textView);
 		sharedPreferences = getSharedPreferences(SETTINGS, Context.MODE_PRIVATE);
+		checkAppVersion();
```
Comment (srifqi) [inline android/app/src/main/java/net/minetest/minetest/MainActivity.java:101]:
```suggestion
```

File: android/app/src/main/java/net/minetest/minetest/MainActivity.java [hunk 2]
```diff
@@ -100,57 +97,10 @@ public void onCreate(Bundle savedInstanceState) {
...
-			checkAppVersion();
-		}
-	}
-
 	private void checkAppVersion() {
-		if (!Environment.getExternalStorageState().equals(Environment.MEDIA_MOUNTED)) {
-			Toast.makeText(this, R.string.no_external_storage, Toast.LENGTH_LONG).show();
-			finish();
-			return;
-		}
```
Comment (srifqi) [inline android/app/src/main/java/net/minetest/minetest/MainActivity.java:104]:
```suggestion
```

File: android/app/src/main/res/values/strings.xml
Comment (srifqi) [inline android/app/src/main/res/values/strings.xml:1]: I am not sure whether we want to remove empty lines here (see other XML files for examples).

File: android/app/src/main/java/net/minetest/minetest/MainActivity.java [hunk 3]
```diff
@@ -100,57 +97,10 @@ public void onCreate(Bundle savedInstanceState) {
...
-					Toast.makeText(this, R.string.not_granted, Toast.LENGTH_LONG).show();
-					finish();
-					return;
-				}
-			}
-			checkAppVersion();
-		}
-	}
-
 	private void checkAppVersion() {
```
Comment (srifqi) [inline android/app/src/main/java/net/minetest/minetest/MainActivity.java:103]:
```suggestion

	private void checkAppVersion() {
```

**Comment by srifqi** [comment]
As a note, 6901c5fae54eafb05494823b60d4e26c14b342f1 is added for the 5.5.0 release (31 January 2022). Any versions older than 5.5.0 needs to update first to any version from 5.5.0 until 5.7.0 for migrating the data files.

**Review by srifqi — CHANGES_REQUESTED** [review]
The concept and its implementation look good to me, except some code style issues.

**Comment by oong819** [comment]
done i think?

File: android/app/src/main/java/net/minetest/minetest/MainActivity.java
Comment (srifqi) [inline android/app/src/main/java/net/minetest/minetest/MainActivity.java:1]: There are some imports that can be removed.
```diff
diff --git a/android/app/src/main/java/net/minetest/minetest/MainActivity.java b/android/app/src/main/java/net/minetest/minetest/MainActivity.java
index 18475b2f8..d942e93e3 100644
--- a/android/app/src/main/java/net/minetest/minetest/MainActivity.java
+++ b/android/app/src/main/java/net/minetest/minetest/MainActivity.java
@@ -25,24 +25,14 @@ import android.content.Context;
 import android.content.Intent;
 import android.content.IntentFilter;
 import android.content.SharedPreferences;
-import android.content.pm.PackageManager;
-import android.os.Build;
 import android.os.Bundle;
-import android.os.Environment;
 import android.view.View;
 import android.widget.ProgressBar;
 import android.widget.TextView;
 import android.widget.Toast;

-import androidx.annotation.NonNull;
 import androidx.annotation.StringRes;
 import androidx.appcompat.app.AppCompatActivity;
-import androidx.core.app.ActivityCompat;
-import androidx.core.content.ContextCompat;
-
-import java.util.ArrayList;
-import java.util.Arrays;
-import java.util.List;

 import static net.minetest.minetest.UnzipService.*;

```

File: android/app/src/main/java/net/minetest/minetest/UnzipService.java
Comment (srifqi) [inline android/app/src/main/java/net/minetest/minetest/UnzipService.java:1]: There is an import that can be removed.
```diff
diff --git a/android/app/src/main/java/net/minetest/minetest/UnzipService.java b/android/app/src/main/java/net/minetest/minetest/UnzipService.java
index 83525a6c5..8c293fa89 100644
--- a/android/app/src/main/java/net/minetest/minetest/UnzipService.java
+++ b/android/app/src/main/java/net/minetest/minetest/UnzipService.java
@@ -28,7 +28,6 @@ import android.app.PendingIntent;
 import android.content.Context;
 import android.content.Intent;
 import android.os.Build;
-import android.os.Environment;
 import android.util.Log;

 import androidx.annotation.NonNull;

```

**Review by srifqi — CHANGES_REQUESTED** [review]
... just a little bit more.

**Review by srifqi — APPROVED** [review]
This PR looks good to me.

**Review by sfan5 — APPROVED** [review]
Looks good too, but someone else should give it a quick test!

**Comment by grorp** [comment]
I think these should be removed too:

https://github.com/minetest/minetest/blob/663c105eea90e99a99a3d115a80634ee3a777788/android/app/src/main/AndroidManifest.xml#L11-L15

https://github.com/minetest/minetest/blob/663c105eea90e99a99a3d115a80634ee3a777788/android/app/src/main/AndroidManifest.xml#L21

**Comment by oong819** [comment]
i wish Minetest ca export and import data files

**Comment by oong819** [comment]
merge pls?

**Pull request merged**